### PR TITLE
Allow for custom css sheet overrides for reports

### PR DESCRIPTION
**Pull request by jchorl**
Adds the ability to specify CSS overrides. I know it's possible to add css to custom modules, but we'd like to do it for the simple/default reports, without specifying entirely new modules.

For example, I'd like to add css rules for the [css `print` media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media). I will merge these with upstream. But for the short term, it's useful to just specify a bunch of overrides, and then get them merged. This will allow for e.g. beautiful print exports of multiqc reports (currently e.g. background colours don't show up). Let me know what you think!

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` has been updated

I tried this out with a few configurations:

```console
$ cat sample.css 
.side-nav {
  width: 800px;
}
$ cat other_sample.css 
.side-nav h1 img {
  height: 200px;
}
$ cat multiqc_config.yaml 
custom_css_files:
  - sample.css
  - other_sample.css
```
Screenshot 2021-11-02 at 03-45-51 MultiQC Report

You can see the logo and sidebar are huge here.

Then I built/ran with cli flags:
```console
$ docker run -it --rm -v $(pwd):/work -w /work jchorl/multiqc --custom-css-files sample.css --custom-css-files other_sample.css .

  /// MultiQC 🔍 | v1.12.dev0

|           multiqc | Search path : /work
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 3/3  
|    custom_content | plasmid_table: Found 2 samples (table)
|           multiqc | Compressing plot data
|           multiqc | Report      : multiqc_report.html
|           multiqc | Data        : multiqc_data
|           multiqc | MultiQC complete
 j@bike  ~/third-party/MultiQC/testdata   jchorl/customcss ±  ls multiqc_config.yaml
ls: cannot access 'multiqc_config.yaml': No such file or directory
```
Screenshot 2021-11-02 at 03-48-03 MultiQC Report

Finally, I removed the flags and config file:
Screenshot 2021-11-02 at 03-49-47 MultiQC Report

We're back to normal looking reports.

### CODE REVIEW

**Comment by jchorl** [comment]
@ewels any chance I could get you to take a look at this? Happy to revise as you see helpful.

File: multiqc/templates/default/includes.html
Comment (ewels) [inline multiqc/templates/default/includes.html:36]: I forget, what is the second argument for? Is it needed?

File: multiqc/multiqc.py
Comment (ewels) [inline multiqc/multiqc.py:452]: I have a feeling that if a click argument accepts multiple, it returns an empty list of not specified. So you can probably ditch this line if that's true.

But worth checking as this is off the top of my head 😅

File: multiqc/multiqc.py
Comment (ewels) [inline multiqc/multiqc.py:180]: I don't think that you need this twice..

File: multiqc/multiqc.py
Comment (ewels) [inline multiqc/multiqc.py:185]: One per use of the flag, so I prefer singular here. 

```suggestion
    help="Custom CSS file to add to the final report",
```

File: CHANGELOG.md
Comment (ewels) [inline CHANGELOG.md:33]:
```suggestion
  - Added support for `--custom-css-files` / `config.custom_css_files` option to include custom CSS in the final report
```

File: docs/config.md
Comment (ewels) [inline docs/config.md:515]:
```suggestion
MultiQC generates HTML reports. You can include custom CSS in your final report if you wish.
```

File: docs/config.md
Comment (ewels) [inline docs/config.md:516]:
```suggestion
Simply add CSS files to the `custom_css_files` config option:
```

File: docs/config.md
Comment (ewels) [inline docs/config.md:523]:
```suggestion
Or pass `--custom-css-files` (can be specified multiple times) and MultiQC will include
```

File: docs/config.md
Comment (ewels) [inline docs/config.md:524]:
```suggestion
them in the final report HTML.
```

**Review by ewels — APPROVED** [review]
Yeah, sorry - I've been on paternity leave and now working 50% so I'm very behind on MultiQC maintenance work.

Love the idea of this PR though, have done a quick read on my phone with one or two minor comments but happy to merge once resolved 👍🏻

Thanks for this!

File: multiqc/templates/default/includes.html
Comment (jchorl) [reply multiqc/templates/default/includes.html:36]:
```
    def include_file(name, fdir=tmp_dir, b64=False):
        try:
            if fdir is None:
                fdir = ""
            if b64:
                with io.open(os.path.join(fdir, name), "rb") as f:
                    return base64.b64encode(f.read()).decode("utf-8")
            else:
                with io.open(os.path.join(fdir, name), "r", encoding="utf-8") as f:
                    return f.read()
```

It's the root dir, looks like. I suspect we can actually omit it. I just copy/pasta'd from right above it.

File: multiqc/multiqc.py
Comment (jchorl) [reply multiqc/multiqc.py:452]: Will test!

File: multiqc/multiqc.py
Comment (jchorl) [reply multiqc/multiqc.py:180]: You're probably right. Above:
```
@click.option(
    "--cl-config", "--cl_config", type=str, multiple=True, help="Specify MultiQC config YAML on the command line"
)
```

But lemme try taking it out :)

**Comment by jchorl** [comment]
Congrats on the pat leave :)

I implemented your suggestions. Thanks for working through.

If you have any additional cycles, I cut another 3 PRs. These 2 should be a slam dunk:
- https://github.com/ewels/MultiQC/pull/1575
- https://github.com/ewels/MultiQC/pull/1576

And this one's a tad hairier:
- https://github.com/ewels/MultiQC/pull/1577

**Comment by ewels** [comment]
Nice, thanks! Ready to merge apart from the failing CI tests that are due to changing CSPs. I never completely wrapped my head around this stuff. Are you ok to look into it and try to get the tests passing?

**Comment by ewels** [comment]
Ah crap, looks like that test is failing on all PRs. Ok may need some thought, I'd be nice to have some green ticks before merging if possible..

**Comment by jchorl** [comment]
Hey @ewels I actually fixed CI in https://github.com/ewels/MultiQC/pull/1578

Let's merge that, and then I can rebase the other PRs, get CI green, $$$?

**Comment by ewels** [comment]
Ah you're a hero! No need to rebase, can just rerun actions.

File: multiqc/multiqc.py
Comment (ewels) [reply multiqc/multiqc.py]: Yeah, past-me probably didn't know better. Only needed when doing short form + long form if I remember correctly.

**Comment by jchorl** [comment]
Seems like actions run at the current sha, not rebasing on master. Should I just rebase and push?

**Comment by ewels** [comment]
Ok fab, thanks!

**Comment by ewels** [comment]
It's none of my business, but how come an engineer from Stripe is running MultiQC? Do you have some bioinformatics side project or are you using MultiQC for something totally different?

**Comment by jchorl** [comment]
> It's none of my business, but how come an engineer from Stripe is running MultiQC? Do you have some bioinformatics side project or are you using MultiQC for something totally different?

Nothing employer/work related - just nerding out on weekends, trying to learn :P

**Comment by ewels** [comment]
Awesome. Happy to have you here 😀

**Pull request merged**
